pkg/registration/product.go

```diff
@@ -102,3 +171,56 @@ func FetchProductInfo(conn connection.Connection, identifier, version, arch stri
 
 	return &product, nil
 }
+
+// Updates the given product (i.e. upgrades/downgrades it) depending on what the
+// SCC server determines. An SCC service is given in return.
+func UpdateProduct(conn connection.Connection, product Product) (Service, error) {
```

Comment:
Can you add tests here? Within the library I think everything should be tested to be in a good spot

Reply:
Also why not call it `Upgrade` rather than `Update`. I think this would be more speaking of what the method does?